python/ray/util/actor_pool.py

```diff
@@ -1,9 +1,10 @@
 from typing import List, Callable, Any
 
 import ray
-from ray.util.annotations import PublicAPI
+from ray.util.annotations import Deprecated, PublicAPI
 
 
+@Deprecated
```

Comment:
Are we not providing a replacement for this?

Reply:
The usage for this API is quite common...

Reply:
We do have a replacement and are recommending users to ray.util.multiprocessing, @ericl

Reply:
I think there is a missing feature in ray.util.multiprocessing that's supported in ActorPool which is stateful actors. Have we discussed that? It's mentioned in the api audit doc.

Reply:
We support stateful actors in Datasets / with BatchPredictor. In general, I don't think we want to support such low-level APIs which are very difficult to use correctly, when users really should use the higher-level ones.

Reply:
The real usage I'm aware of is
```
@ray.remote()
class Predictor:
def _init_(self, model_path: str):
# Load model
self.model = tf.keras.models.load_model(model_path)

def predict(self, input_file_path: str, output_path: str):
# predicate
self.model.predicate(...)

predictors = [
Predictor.options(num_cpus=cpus_per_actor).remote(model_path)
for i in range(actor_count)
]
pool = ActorPool(predictors)
list(
pool.map_unordered(
lambda pc, file_path: pc.predict.remote(file_path, output_path), dataset_partitions
)
)
```
For this particular one, I think we should recommend Datasets with BatchPredictor.

But what if there is non-ML use case? There is no higher-level apis for them.

Reply:
@ericl @richardliaw what's your thoughts on this?

Reply:
I suggest we try deprecating but see how the community reacts.

I would also suggest that if X is deprecated that we explicitly tell the community that we are open to feedback and that they should reach out to us on github.